Sources/NIOSSL/SSLCertificateName.swift

```diff
@@ -0,0 +1,81 @@
+@_implementationOnly import CNIOBoringSSL
+
+public struct SSLCertificateNameType: Equatable, Hashable, Sendable {
+    public var nid: Int32
+    public static let organization = SSLCertificateNameType(nid: NID_organizationName)
+    public static let organizationalUnit = SSLCertificateNameType(nid: NID_organizationalUnitName)
+    public static let state = SSLCertificateNameType(nid: NID_stateOrProvinceName)
+    public static let country = SSLCertificateNameType(nid: NID_countryName)
+    public static let city = SSLCertificateNameType(nid: NID_localityName)
+    public static let commonName = SSLCertificateNameType(nid: NID_commonName)
+    public static let emailAddress = SSLCertificateNameType(nid: NID_pkcs9_emailAddress)
+    public static let userId = SSLCertificateNameType(nid: NID_userId)
+}
+
+public struct SSLCertificateName: Equatable, Hashable, Sendable {
+    public var value: String
+    public var type: SSLCertificateNameType
+
+    public init(_ value: String, _ type: SSLCertificateNameType) {
+        self.value = value
+        self.type = type
+    }
+}
+
+extension NIOSSLCertificate {
+    private static func convertName(_ name: OpaquePointer) -> [SSLCertificateName] {
+
+        let count = CNIOBoringSSL_X509_NAME_entry_count(name)
+        var names = [SSLCertificateName]()
+        names.reserveCapacity(Int(count))
+        for index in 0..<count {
+            guard let entry = CNIOBoringSSL_X509_NAME_get_entry(name, index) else {
+                continue
+            }
+
+            guard let object = CNIOBoringSSL_X509_NAME_ENTRY_get_object(entry) else {
+                continue
+            }
+
+            guard let data = CNIOBoringSSL_X509_NAME_ENTRY_get_data(entry) else {
+                continue
+            }
+
+            var encodedName: UnsafeMutablePointer<UInt8>? = nil
+            let stringLength = CNIOBoringSSL_ASN1_STRING_to_UTF8(&encodedName, data)
+
+            guard let namePtr = encodedName else {
+                continue
+            }
+            let arr = UnsafeBufferPointer(start: namePtr, count: Int(stringLength))
+            let nameString = String(decoding: arr, as: UTF8.self)
+            CNIOBoringSSL_OPENSSL_free(namePtr)
```

Comment:
Mild nit: it's often nicer to stick this in a `defer` immediately after the `guard`. That way we know we can't possibly forget this in a refactor.